packages/babel-generator/src/printer.ts

```diff
@@ -865,12 +865,16 @@ class Printer {
 
   printJoin(
     nodes: Array<t.Node> | undefined | null,
-    opts: PrintJoinOptions = {},
+    statement?: boolean,
+    indent?: boolean,
+    separator?: PrintJoinOptions["separator"],
+    printTrailingSeparator?: boolean,
+    addNewlines?: PrintJoinOptions["addNewlines"],
+    iterator?: PrintJoinOptions["iterator"],
+    trailingCommentsLineOffset?: number,
   ) {
```

Comment:
Their order is similar to the frequency of use.

Reply:
There are quite a few boolean options here, maybe we can pack them into bit flags.

Reply:
I was originally working on bit flags, but then realized the code needed was pretty much the same either way.
I'll switch it over to bit flags and see if that makes it a bit cleaner. :)

Reply:
Seems a bit difficult since `indent===null` would make sense.